server/channels/api4/report.go

```diff
@@ -85,3 +87,43 @@ func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Logger.Warn("Error writing response", mlog.Err(jsonErr))
 	}
 }
+
+func getUserCountForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
+		c.SetPermissionError(model.PermissionSysconsoleReadUserManagementUsers)
+		return
+	}
+
+	teamFilter := r.URL.Query().Get("team_filter")
```

Comment:
The one in `getUsersForReporting` adds a lot around the paging, which the user count does not need. I wanted to make use of the same object in order to avoid making a bunch more definitions, but they are filled differently.

We can merge them if you really want, but it would lead to the confusion that the count API supports the paging stuff, which it doesn't.

Reply:
The difference is the one in `getUsersForReporting` fills `ReportingBaseOptions`, and the one in count doesn't. Except those 5 lines it s a copy-paste code, which we need to avoid. You can create a function that all we're doing here from line 97 to 118 and `getUsersForReporting` can fill the base options on the options object returned from this function. There is a large enough, seperate enough chunk of code copy-paste between the two, so we should consider having a common function.

Reply:
The difference is the one in `getUsersForReporting` fills `ReportingBaseOptions`, and the one in count doesn't. Except those 5 lines it s a copy-paste code, which we need to avoid. You can create a function that all we're doing here from line 97 to 118 and `getUsersForReporting` can fill the base options on the options object returned from this function. There is a large enough, separate enough chunk of code copy-paste between the two, so we should consider having a common function.

Reply:
The difference is the one in `getUsersForReporting` fills `ReportingBaseOptions`, and the one in count doesn't. Except those 5 lines it s a copy-paste code, which we need to avoid. You can create a function that all we're doing here from line 97 to 118 and `getUsersForReporting` can fill the base options on the options object returned from this function. There is a large enough, separate enough chunk of code copy-paste between the two, so we should consider having a common function.

Reply:
Okay I've refactored that